internal/pkg/template/override/override.go

```diff
@@ -0,0 +1,39 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package override renders the manifest override rules to the CloudFormation template.
+package override
+
+// OverrideCloudFormationTemplate overrides the given CloudFormation template by applying
+// the override rules.
+func OverrideCloudFormationTemplate(overrideRules []string, origTemp string) (string, error) {
+	content, err := unmarshalCFNYaml(origTemp)
+	if err != nil {
+		return "", err
+	}
+	rules, err := parseRules(overrideRules)
+	if err != nil {
+		return "", err
+	}
+	destContent, err := applyRulesToCFNTemplate(rules, content)
+	if err != nil {
+		return "", err
+	}
+	output, err := marshalCFNYaml(destContent)
+	if err != nil {
+		return "", err
+	}
+	return output, nil
+}
+
+func unmarshalCFNYaml(temp string) (map[interface{}]interface{}, error) {
```

Comment:
i thought about that! i'll sub those "!" with keyword in when unmarshaling then sub them back when marshaling.

Reply:
`*yaml.Node` won't be too useful when we use `reflect` to apply those overriding rules.

Reply:
OK cool, I like this second revision is a good starting point!

Is there a particular reason why you prefer using `reflect` over `yaml`?

>  i'll sub those "!" with keyword in when unmarshaling then sub them back when marshaling.  

I am slightly worried about the fragility of this approach like what if they have a string that has "!Sub" or something? 💭 This might be just my familiarity over refect, but I think`*yaml.Node` would provide us a more robust and easy way of overriding values.

For example:
```go
// Here we are swapping the HardLimit value
for i:= 0; i < len(n.Content); i += 2 {
   if n.Content[i].Value != "HardLimit"  {
        continue
   }
   n.Content[i + 1] = newValueNode
}
```
Let me know what you think!

Reply:
shameless plug https://www.efekarakus.com/2020/05/30/deep-dive-go-yaml-cfn.html 😂

Reply:
discussed offline. Let's do it with `yaml` https://play.golang.org/p/lcztaKZSjQC